ui: Link from statement diagnostics to details page

koorosh opened the pull request:
Resolves: #46559

We have Statements diagnostics history page with list
of all requested diagnostics.
Before, statement fingerprint were represented as a
simple text and now it is a links to statement details
page.

One notion, that it is possible to have diagnostics for
statements which is already cleared. In this case
statement is displayed as a text instead of link.

Release note (admin ui change): Add links to statement
details from Statement Diagnostics history page.

Release justification: bug fixes and low-risk updates to new functionality

![out](https://user-images.githubusercontent.com/3106437/78255335-d824e300-74ff-11ea-9203-233ac8ba67fc.gif)

koorosh committed:
ui: Link from statement diagnostics to details page

We have Statements diagnostics history page with list
of all requested diagnostics.
Before, statement fingerprint were represented as a
simple text and now it is a links to statement details
page.

One notion, that it is possible to have diagnostics for
statements which is already cleared. In this case
statement is displayed as a text instead of link.

Release note (admin ui change): Add links to statement
details from Statement Diagnostics history page.

Release justification: bug fixes and low-risk updates to new functionality

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/46923)
<!-- Reviewable:end -->

koorosh commented:
`do-not-merge` Waiting for answer: https://github.com/cockroachdb/cockroach/issues/46559#issuecomment-607828901

koorosh committed:
ui: Show tooltip with full statement in Diagnostics history page

Before, statements column on Statements diagnostics history page
displayed full statement and didn't have tooltips.

With current changes, statements column behave almost the same
way as on Statements page:
- statement is shortened
- tooltip is added with full statement
- in case shortened statement is the same as full statement -
then don't display tooltip at all.

Release note (admin ui change): Add tooltips with full length
statements on Statement diagnostics history page.

Release justification: bug fixes and low-risk updates to new functionality

koorosh commented:
Additional change is applied, now statements column looks and behave almost the same
way as on Statements page:
- statement is shortened
- tooltip is added with full statement
- in case shortened statement is the same as full statement -
then don't display tooltip at all.

![out 12 46 08](https://user-images.githubusercontent.com/3106437/78349432-4921d480-75ac-11ea-9d94-0f04cb608213.gif)

dhartunian reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46923)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @koorosh and @nathanstilwell)

___
*[pkg/ui/src/views/reports/containers/statementDiagnosticsHistory/index.tsx, line 79 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46923#-M4Q4rhM5a6ZyayaEU2L:-M4Q4rhM5a6ZyayaEU2M:b-xdv8b1) ([raw file](https://github.com/cockroachdb/cockroach/blob/9ad46489cd2c4d9d704c9f37e028962476a5cd38/pkg/ui/src/views/reports/containers/statementDiagnosticsHistory/index.tsx#L79)):*
> ```XML
>         return (
>           <Link
>             to={ `/statement/${implicitTxn}/${encodeURIComponent(query)}` }
> ```

Given the fact that this will be used outside of this repo, can we extract the creation of this link?


<!-- Sent from Reviewable.io -->

koorosh commented:
> can we extract the creation of this link

I would consider this as a separate more global task, probably in scope of routes and the way we provide/consume configuration.

dhartunian reviewed (approved):
Appreciate the comment! We can address later.

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46923#-:-M4VhgEYDYldf18uxCCX:b-4nijo0)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @koorosh and @nathanstilwell)



<!-- Sent from Reviewable.io -->

dhartunian commented:
bors r+
